### PR TITLE
[Merged by Bors] - feat: Port GroupTheory.GroupAction.Prod

### PR DESCRIPTION
**Pull request by adamtopaz**
mathlib3 SHA: aba57d4d3dae35460225919dcd82fe91355162f9

### CODE REVIEW

File: Mathlib/GroupTheory/GroupAction/Prod.lean [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+-/
+import Mathlib.Algebra.Group.Prod
+import Mathlib.GroupTheory.GroupAction.Defs
+
+/-!
+# Prod instances for additive and multiplicative actions
+This file defines instances for binary product of additive and multiplicative actions and provides
+scalar multiplication as a homomorphism from `α × β` to `β`.
+## Main declarations
+* `smul_mul_hom`/`smul_monoid_hom`: Scalar multiplication bundled as a multiplicative/monoid
```
Comment (semorrison) [inline Mathlib/GroupTheory/GroupAction/Prod.lean:18]:
```suggestion
* `smulMulHom `/`smulMonoidHom `: Scalar multiplication bundled as a multiplicative/monoid
```

File: Mathlib/GroupTheory/GroupAction/Prod.lean [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  homomorphism.
+## See also
+* `group_theory.group_action.option`
+* `group_theory.group_action.pi`
+* `group_theory.group_action.sigma`
+* `group_theory.group_action.sum`
```
Comment (semorrison) [inline Mathlib/GroupTheory/GroupAction/Prod.lean:24]:
```suggestion
* `Mathlib.GroupTheory.GroupAction.option`
* `Mathlib.GroupTheory.GroupAction.Pi`
* `Mathlib.GroupTheory.GroupAction.Sigma`
* `Mathlib.GroupTheory.GroupAction.Sum`
```

**Comment by semorrison** [comment]
bors d+

Please change `Yes` to `No` on the port status wiki page when you merge.

File: Mathlib/GroupTheory/GroupAction/Prod.lean [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+/-!
+# Prod instances for additive and multiplicative actions
+This file defines instances for binary product of additive and multiplicative actions and provides
+scalar multiplication as a homomorphism from `α × β` to `β`.
+## Main declarations
+* `smulMulHom `/`smulMonoidHom `: Scalar multiplication bundled as a multiplicative/monoid
+  homomorphism.
+## See also
+* `Mathlib.GroupTheory.GroupAction.option`
```
Comment (adamtopaz) [inline Mathlib/GroupTheory/GroupAction/Prod.lean:21]:
```suggestion
* `Mathlib.GroupTheory.GroupAction.Option`
```

**Pull request closed without merging**